app/models/mixins/ownership_mixin.rb

```diff
@@ -110,6 +110,6 @@ def owned_by_current_ldap_group
   end
 
   def set_tenant_from_group
-    self.tenant_id = miq_group.tenant_id if miq_group
+    self.tenant_id = miq_group.tenant_id if miq_group_id_changed? && miq_group
```

Comment:
1. I wonder if we want to update the record if either the group or the tenant changes.
2. I think we cover most of these cases but could you verify that we are testing:
  - create an object with no group
  - create an object with an `miq_group`
  - create an object with an `miq_group_id`
  - change a miq_group from nil to value
  - change a miq_group from value to nil

3. Could you test the root premise.

```ruby
r = record.create().reload
expect { r.save }.not_to access_database
```

If the root premise does a couple of saves, you don't need to fix it.
Instead just try for reducing the no-op query count by 1